File: 1512.number-of-good-pairs.java
import java.util.HashMap;
import java.util.Map;

/*
 * @lc app=leetcode id=1512 lang=java
 *
 * [1512] Number of Good Pairs
 */

// @lc code=start
class Solution {
    public int numIdenticalPairs(int[] nums) {
        Map<Integer, Integer> seen_count = new HashMap<>();
        int res = 0;
        for (int n : nums) {
            res += seen_count.getOrDefault(n, 0);
            seen_count.put(n, seen_count.getOrDefault(n, 0) + 1);
        }
        return res;
    }
}
// @lc code=end

File: 1512.number-of-good-pairs.py
#
# @lc app=leetcode id=1512 lang=python3
#
# [1512] Number of Good Pairs
#

# @lc code=start
from collections import Counter


class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        seen_count = Counter()
        res = 0
        for n in nums:
            res += seen_count[n]
            seen_count[n] += 1
        return res
# @lc code=end
